### DISCUSSION TITLE
Microphone Usage for Sense Version | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#685] [Q&A]
# Microphone Usage for Sense Version | Seeed Studio Wiki

This article describes how to use the microphone on the XIAO ESP32S3 Sense expansion board.

https://wiki.seeedstudio.com/xiao_esp32s3_sense_mic/

**Comment by fjmsouza** [comment]
Hello, 
I'm achieving 0(zero) value always in I2S.read(). What would I do?

**Comment by fjmsouza** [comment]
Solved.
Some FW problem. Solved through workaround:
-Updating IDE to Version: 2.2.1
Date: 2023-08-31T14:35:44.802Z
CLI Version: 0.34.0

And following the tutorial https://wiki.seeedstudio.com/xiao_esp32s3_getting_started/

Obs.: perhaps this clear the previous conflicting preferences/settings.

**Comment by beslagic** [comment]
For Windows 10 and 11 users, use Rufus to format the SD card, as other applications may cause issues with the SD.begin(21) function, such as 'Failed to mount SD card!' or files not being visible on the SD card when viewed on a PC.

**Comment by feiticeir0** [comment]
Is there a way to increase the audio volume ? Not in playback, but when recording ?  The sound is faint..

**Reply by nfs0619** [reply]
Due to hardware limitations, it may not be possible to directly increase the volume during recording. However, you might want to try using software gain. By applying a gain factor (e.g., 2x gain) to the recorded audio samples, you can increase the audio signal's volume after it's captured. Just be cautious, as too much gain can cause distortion, so make sure to limit the range to avoid clipping.

If you have any further questions, I recommend reaching out to our technical support team at [techsupport@seeed.io](mailto:techsupport@seeed.io).

**Comment by migcruz** [comment]
just wondering if you have a sample with zephyr 4.3? Been trying to get the microphone to work but all I'm getting is white noise with very high RMS2 values.

**Reply by Lukilyy** [reply]
Hello, we currently do not have the relevant sample code for Zephyr 4.3. We only provide usage examples in the Arduino IDE environment. We recommend that you first use the Arduino example code in the documentation for programming and testing. If the Arduino environment runs normally, you can confirm that the hardware is correct, and the problem mainly lies in the driver configuration of Zephyr. This microphone uses the PDM protocol. Please ensure that your configuration is consistent with the hardware: CLK is GPIO 42, and DATA is GPIO 41.
https://wiki.seeedstudio.com/xiao_esp32s3_sense_mic/#getting-started
